RecoLocalCalo/EcalRecProducers/plugins/TimeComputationKernels.cu

```diff
@@ -0,0 +1,1143 @@
+                                                  SampleVector::Scalar const* sample_value_errors,
+                                                  uint32_t const* dids_eb,
+                                                  uint32_t const* dids_ee,
+                                                  bool const* useless_sample_values,
+                                                  char const* pedestal_nums,
+                                                  ConfigurationParameters::type const* amplitudeFitParametersEB,
+                                                  ConfigurationParameters::type const* amplitudeFitParametersEE,
+                                                  ConfigurationParameters::type const* timeFitParametersEB,
+                                                  ConfigurationParameters::type const* timeFitParametersEE,
+                                                  SampleVector::Scalar const* sumAAsNullHypot,
+                                                  SampleVector::Scalar const* sum0sNullHypot,
+                                                  SampleVector::Scalar* tMaxAlphaBetas,
+                                                  SampleVector::Scalar* tMaxErrorAlphaBetas,
+                                                  SampleVector::Scalar* g_accTimeMax,
+                                                  SampleVector::Scalar* g_accTimeWgt,
+                                                  TimeComputationState* g_state,
+                                                  unsigned const int timeFitParameters_sizeEB,
+                                                  unsigned const int timeFitParameters_sizeEE,
+                                                  ConfigurationParameters::type const timeFitLimits_firstEB,
+                                                  ConfigurationParameters::type const timeFitLimits_firstEE,
+                                                  ConfigurationParameters::type const timeFitLimits_secondEB,
+                                                  ConfigurationParameters::type const timeFitLimits_secondEE,
+                                                  const int nchannels,
+                                                  uint32_t const offsetForInputs) {
+      using ScalarType = SampleVector::Scalar;
+
+      // constants
+      constexpr int nthreads_per_channel = 45;  // n=10, n(n-1)/2
+      constexpr int nsamples = EcalDataFrame::MAXSAMPLES;
+
+      // indices
+      const int gtx = threadIdx.x + blockDim.x * blockIdx.x;
+      const int ch = gtx / nthreads_per_channel;
+      const int ltx = threadIdx.x % nthreads_per_channel;
+      const int ch_start = ch * nsamples;
+      const auto* dids = ch >= offsetForInputs ? dids_ee : dids_eb;
+      const int inputCh = ch >= offsetForInputs ? ch - offsetForInputs : ch;
+
+      // rmeove inactive threads
+      // TODO: need to understand if this is 100% safe in presence of syncthreads
```

Comment:
Should this TODO be addressed?

Reply:
According to @fwyzard this should be fine since threads that have exited the kernel shouldn't matter. I will update the comment.

Reply:
my comment is that it should be addressed by inverting the logic of the if block into an early return - not left as is

Reply:
In this case here this as actually already in the early return form you have proposed.